libpromises/verify_classes.c

```diff
@@ -63,20 +63,20 @@ PromiseResult VerifyClassPromise(EvalContext *ctx, const Promise *pp, ARG_UNUSED
         xsnprintf(pp->promiser, strlen(pp->promiser) + 1, "%s", CanonifyName(pp->promiser));
     }
 
-    if (a.context.nconstraints == 0)
-    {
-        cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_FAIL, pp, a, "No constraints for class promise '%s'", pp->promiser);
-        return PROMISE_RESULT_FAIL;
-    }
-
     if (a.context.nconstraints > 1)
     {
         cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_FAIL, pp, a, "Irreconcilable constraints in classes for '%s'", pp->promiser);
         return PROMISE_RESULT_FAIL;
     }
 
-    if (EvalClassExpression(ctx, a.context.expression, pp))
+    if (NULL == a.context.expression ||
```

Comment:
@tzz minor note, since you are a main contributor and your code is all over the place, please try to follow the coding standard form next time, it doesn't help having two styles in the same project.